yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -282,6 +282,16 @@ contract NonNativeToken {
         secretHash: Field,
         _padding: [Field; abi::MAX_ARGS - 2]
     ) {
+        // Decrease user's balance.
+        let sender = inputs.call_context.msg_sender;
+        let sender_balance = public_balances.at(sender);
+        let current_sender_balance = sender_balance.read();
+
+        if (current_sender_balance as u120) > (amount as u120) {
```

Comment:
When `amount >= current_sender_balance` seems like I would just create the commitment without spending any funds 😬 

We should assert that `current_sender_balance <= amount` and then we can decrease afterwards.

Reply:
ah good spot, this was lazy